spyder/utils/programs.py

```diff
@@ -138,6 +138,19 @@ def alter_subprocess_kwargs_by_platform(**kwargs):
         # We "or" them together
         CONSOLE_CREATION_FLAGS |= CREATE_NO_WINDOW
         kwargs.setdefault('creationflags', CONSOLE_CREATION_FLAGS)
+
+        # ensure Windows subprocess environment has SYSTEMROOT, but variable
+        # is case insensitive
+        if ((kwargs.get('env') is not None) and
+            ('SYSTEMROOT' not in map(str.upper, kwargs['env'].keys()))):

```

Comment:
This is very convoluted. Please move the detection of `SYSTEMROOT`among the env vars to a local variable above this `if`.

Reply:
Okay, will do.

Reply:
I must check that ``env`` is in ``kwargs`` first, so I cannot move the check for ``SYSTEMROOT`` above the ``if`` statement. But I tried to clean up the code so that it was more readable.